src/Datadog.Trace/Configuration/ConfigurationSourceExtensions.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Configuration
+{
+    internal static class ConfigurationSourceExtensions
+    {
+        private const char Separator = ';';
+
+        public static IEnumerable<string> GetStrings(this IConfigurationSource source, string key)
+        {
+            return source?.GetString(key)
+                ?.Split(new[] { Separator }, StringSplitOptions.RemoveEmptyEntries)
+                ?? Enumerable.Empty<string>();
+        }
+
+        public static TEnum GetTypedValue<TEnum>(this IConfigurationSource source, string key)
+            where TEnum : struct, IConvertible
+        {
+            Enum.TryParse(source?.GetString(key), ignoreCase: true, out TEnum typedValue);
+            return typedValue;
+        }
+
+        public static IEnumerable<TEnum> GetTypedValues<TEnum>(this IConfigurationSource source, string key)
```

Comment:
how we would know that an invalid value was provided. [more context](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/102#discussion_r607774012)

Reply:
I'm not sure this is the right place to deal with validation. As mentioned in #102 it should fail fast. I would propose configuration source validator in early startup (binding configuration to settings instance). After that point configuration should be valid for whole application.

Reply:
For now, would you consider making this use `Enum.Parse` and throw an exception so we know that the result is valid? I'd prefer that this PR close with a clear indication of configuration errors, so this change might be easier than doing a configuration source validator right now

Reply:
Originally I tried not to change parsing logic with this PR (notice the previous logic is 1:1 same as now) but I can easily change it there is interest.

Reply:
I suggest doing it separately